### PR TITLE
Fix issues with tagging/pushing/pulling

### PR DESCRIPTION
**Pull request by nre-ableton**
This PR fixes two major issues with groovylint:

- It now uses the build number as the `build -t <TAG>` tag name when building, which fixes a bug where a cached image from a previous build with the same name would be tagged instead of the one from this build.
- It only calls `docker pull` in the pipeline library when the image is *not* specified. This fixes a bug where, again, a cached image could be used in place of the correctly built image when testing the library.

Fixes https://github.com/AbletonAppDev/devtools/issues/1337, ping @AbletonDevTools/gotham-city (btw take note of the awesome issue number 🕶 💻)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -31,7 +31,7 @@ runTheBuilds.runDevToolsProject(
     data['venv'] = venv
   },
   build: { data ->
-    data['image'] = docker.build('abletonag/groovylint')
+    data['image'] = docker.build("abletonag/groovylint:${env.BUILD_NUMBER}")
```
[the remark this answers is not in the document]
Comment (nre-ableton) [reply Jenkinsfile]: @ala-ableton Sure, fixed and pushed.

File: Jenkinsfile [hunk 2]
```diff
@@ -31,7 +31,7 @@ runTheBuilds.runDevToolsProject(
     data['venv'] = venv
   },
   build: { data ->
-    String gitHash = sh(returnStdout: true, script: 'git rev-parse HEAD')
+    String gitHash = sh(returnStdout: true, script: 'git rev-parse HEAD')[0..7]
```
Comment (ala-ableton) [inline Jenkinsfile]: @nre-ableton `git rev-parse --short HEAD` please.

File: Jenkinsfile [hunk 3]
```diff
@@ -31,7 +31,7 @@ runTheBuilds.runDevToolsProject(
     data['venv'] = venv
   },
   build: { data ->
-    String gitHash = sh(returnStdout: true, script: 'git rev-parse HEAD')[0..7]
+    String gitHash = sh(returnStdout: true, script: 'git rev-parse --short HEAD')[0..7]
```
Comment (ala-ableton) [inline Jenkinsfile]: @nre-ableton the point of using `--short` is to get rid of the `[0..7]` hack :sweat_smile:

Comment (nre-ableton) [reply Jenkinsfile]: Ah, ok, I didn't realize that. :wink:

Comment (nre-ableton) [reply Jenkinsfile]: @ala-ableton It's worth pointing out that part of the reason I did `[0..7]` is that it spares us a `trim()` call as well...

**Comment by ala-ableton** [comment]
Once rebased, LGTM :+1:

**Pull request merged**